spec/download_spec.rb

```diff
@@ -11,8 +10,12 @@
 
   it "downloads a csv file with given controller and given objects" do
     objs = Product.all
-    # text = @exporter.to_csv(objs)
-    @exporter.download(@products_controller, objs)
-    # @products_controller.expects(:send_data).with(text).returns(:success)
+    text = @exporter.to_csv(objs)
+    opt = {filename: nil}
+
+    allow(@products_controller).to receive(:send_data).with(text, opt).and_return(true)
+
+    result = @exporter.download(@products_controller, objs)
+    expect(result).to eq true
```

Comment:
This test passes because in your `and_return` above you basically tell the function to return true.